packages/x-charts-pro/src/SankeyChart/calculateSankeyLayout.ts

```diff
@@ -67,11 +63,18 @@ export function calculateSankeyLayout(
     .nodeId((d) => d.id)
     .iterations(iterations);
 
-  if (nodeSort) {
-    sankeyGenerator = sankeyGenerator.nodeSort(nodeSort);
+  // For 'auto' or undefined, don't set anything (use d3-sankey default behavior)
+  if (typeof nodeSort === 'function') {
+    sankeyGenerator.nodeSort(nodeSort);
+  } else if (nodeSort === 'fixed') {
+    // Null is not accepted by the types.
+    sankeyGenerator.nodeSort(null as any);
   }
-  if (linkSort) {
-    sankeyGenerator = sankeyGenerator.linkSort(linkSort);
+  if (typeof linkSort === 'function') {
+    sankeyGenerator.linkSort(linkSort);
+  } else if (linkSort === 'fixed') {
+    // Null is not accepted by the types.
```

Comment:
Probably a good opportunity to contribute to the types. 

However, from looking at the code it seems this will be used by the native array's `sort` function. Shouldn't we pass `undefined` instead of `null`? 

I'm not even sure how `null` isn't throwing, because if I try this in the console, it throws: 

```js
const a = [3,2,1]
a.sort(null)
// Uncaught TypeError: The comparison function must be either a function or undefined: null
//    at Array.sort (<anonymous>)
//    at <anonymous>:1:3
```

Reply:
> This is not a direct sort function. We pass it to d3 and they check if it is null/undefined for different behaviours.

You're right. They don't check it in the setter, but when using the sort 👍 

> The d3 type pr is up, but has no review for 2+ weeks

Unfortunate.

Can we link it in the comment above the cast? Just in case we forget to update it